src/xml-properties-mixin.js

```diff
@@ -0,0 +1,151 @@
+const re = require('block-re');
+const cheerio = require('cheerio');
+
+const makeNSTagSelector = require('./make-ns-tag-selector');
+
+const $ = cheerio.load(`<?xml version="1.0" ?>`, { xmlMode: true });
+
+//
+// mixin to persist data in an xml properties element, similar to how properties
+// are stored in pPr/rPr elements in ooxml.
+//
+// numbers and strings are stored as the 'val' attribute of an element, e.g.
+// properties.foo = 5 ==> <foo val="5"/>
+//
+// key: value objects
+// properties.foo = { bar: 5, baz: 'bat' } ==> <foo bar="5", baz="bat"/>
+//
+// array elements
+// properties.push('foo', {bar: 5})
+// properties.push('foo', {bar: 6})
+// ==> <foo bar="5"/><foo bar="6"/>
+//
+
+let XmlPropertiesMixin = (superclass) => class extends superclass {
+  constructor(root, propertiesTag, namespace) {
+    if (!propertiesTag) {
+      throw new Error(`need propertiesTag for XmlProperties`);
+    }
+    super();
+      // accept either an XmlProperties object (with a .$root property) or a dom element
```

Comment:
Unfortunately this pattern is at the core of my vendetta. Polymorphism can add to developer convenience, but subtract a lot from longer-term readability and maintainability, which for this code is a very high priority IMO as a believe it will evolve into widely-used core technology for  our company.

I have always hated that jQuery pattern of accepting just anything you give it. I think it made sense way back when Javascript wasn't really a programming language for software engineers, but just a way for HTML hackers to make their pages a little smarter, but I don't think it makes sense in our context, and when I compare my experience of trying to debug through, for example, most Ember addons or NodeJS libraries compared to trying to debug through jQuery plugins, I get even more pig-headed about my conviction 😛 

If your priority is debug cycles, could you not just put in an assert that makes sure what's passed in is/isn't the kind of thing that you do/don't want so if you pass in the wrong one, you instantly know where the problem is?